src/utils/mod.rs

```diff
@@ -22,3 +23,23 @@ pub fn parse_hashbang(program: &Path) -> Option<String> {
         None
     }
 }
+
+pub fn with_has_more<Element>(
+    into_iter: impl IntoIterator<Item = Element>,
+) -> impl Iterator<Item = (Element, bool)> {
+    struct Iter<Element, I: Iterator<Item = Element>>(Peekable<I>);
+
+    impl<Element, I: Iterator<Item = Element>> Iterator for Iter<Element, I> {
+        type Item = (Element, bool);
+
+        fn next(&mut self) -> Option<Self::Item> {
+            let result = self.0.next();
+            match result {
+                None => None,
+                Some(current) => Some((current, self.0.peek().is_some())),
+            }
+        }
+    }
+
+    Iter(into_iter.into_iter().peekable())
```

Comment:
`into_iter.into_iter()` reads funny, is there a better name for the argument to `with_has_more`?

Reply:
Inside `with_has_more` the only thing we know about the argument is that it implements `IntoIterator`. So I thought `into_iter` was the best name. What would be better? `input`? `iterable`? Or just `iterator`?